llvm/tools/sycl-post-link/sycl-post-link.cpp

```diff
@@ -806,108 +806,96 @@ processInputModule(std::unique_ptr<Module> M) {
-      // when linked back because functions shared between graphs are cloned and
-      // renamed.
-      std::unique_ptr<module_split::ModuleSplitterBase> ESIMDSplitter =
-          module_split::getSplitterByKernelType(std::move(MDesc1),
-                                                EmitOnlyKernelsAsEntryPoints);
-      const bool SplitByESIMD = ESIMDSplitter->remainingSplits() > 1;
-      Modified |= SplitByESIMD;
-
-      if (SplitByESIMD && SplitByScope &&
-          (SplitMode == module_split::SPLIT_PER_KERNEL) && !SplitEsimd) {
-        // Controversial state reached - SYCL and ESIMD entry points resulting
-        // from SYCL/ESIMD split (which is done always) are linked back, since
-        // -split-esimd is not specified, but per-kernel split is requested.
-        warning("SYCL and ESIMD entry points detected and split mode is "
-                "per-kernel, so " +
-                SplitEsimd.ValueStr + " must also be specified");
-      }
-      SmallVector<module_split::ModuleDesc, 2> MMs;
-
-      while (ESIMDSplitter->hasMoreSplits()) {
-        module_split::ModuleDesc MDesc2 = ESIMDSplitter->nextSplit();
-        DUMP_ENTRY_POINTS(MDesc2.entries(), MDesc2.Name.c_str(), 3);
-        Modified |= processSpecConstants(MDesc2);
-
-        // TODO: detach compile-time properties from device globals.
-        if (DeviceGlobals.getNumOccurrences() > 0) {
-          Modified |=
-              runModulePass<CompileTimePropertiesPass>(MDesc2.getModule());
-        }
-        if (!MDesc2.isSYCL() && LowerEsimd) {
-          assert(MDesc2.isESIMD() && "NYI");
-          // ESIMD lowering also detects large-GRF kernels, so it must happen
-          // before large-GRF split.
-          Modified |= lowerEsimdConstructs(MDesc2);
-        }
-        MMs.emplace_back(std::move(MDesc2));
-      }
-      if (!SplitEsimd && (MMs.size() > 1)) {
-        // SYCL/ESIMD splitting is not requested, link back into single module.
-        assert(MMs.size() == 2);
-        assert((MMs[0].isESIMD() && MMs[1].isSYCL()) ||
-               (MMs[1].isESIMD() && MMs[0].isSYCL()));
-        int ESIMDInd = MMs[0].isESIMD() ? 0 : 1;
-        int SYCLInd = MMs[0].isESIMD() ? 1 : 0;
-        // ... but before that, make sure no link conflicts will occur.
-        MMs[ESIMDInd].renameDuplicatesOf(MMs[SYCLInd].getModule(), ".esimd");
-        module_split::ModuleDesc M2 =
-            link(std::move(MMs[0]), std::move(MMs[1]));
-        M2.restoreLinkageOfDirectInvokeSimdTargets();
-        string_vector Names;
-        M2.saveEntryPointNames(Names);
-        M2.cleanup(); // may remove some entry points, need to save/rebuild
-        M2.rebuildEntryPoints(Names);
-        MMs.clear();
-        MMs.emplace_back(std::move(M2));
-        DUMP_ENTRY_POINTS(MMs.back().entries(), MMs.back().Name.c_str(), 3);
-        Modified = true;
-      }
+    MDesc.fixupLinkageOfDirectInvokeSimdTargets();
```

Comment:
Note for reviewers: this huge chunk of changes is essentially a removal of a `while` loop with shifting the body to the left for one indentation level